packages/mui-material/src/styles/createTypography.js

```diff
@@ -72,6 +72,13 @@ export default function createTypography(palette, typography) {
     button: buildVariant(fontWeightMedium, 14, 1.75, 0.4, caseAllCaps),
     caption: buildVariant(fontWeightRegular, 12, 1.66, 0.4),
     overline: buildVariant(fontWeightRegular, 12, 2.66, 1, caseAllCaps),
+    inherit: {
+      fontFamily: 'inherit',
+      fontWeight: 'inherit',
+      fontSize: 'inherit',
+      lineHeight: 'inherit',
+      letterSpacing: 'inherit',
+    },
```

Comment:
I don't understand. This variant doesn't make sense for me in the theme. Why not solve the problem like this in the first place? 

```diff
diff --git a/packages/mui-material/src/Typography/Typography.js b/packages/mui-material/src/Typography/Typography.js
index 1de18d23f7..f5421fc59b 100644
--- a/packages/mui-material/src/Typography/Typography.js
+++ b/packages/mui-material/src/Typography/Typography.js
@@ -43,6 +43,9 @@ export const TypographyRoot = styled('span', {
   },
 })(({ theme, ownerState }) => ({
   margin: 0,
+  ...(ownerState.variant === 'inherit' && {
+    font: 'inherit',
+  }),
   ...(ownerState.variant && theme.typography[ownerState.variant]),
   ...(ownerState.align !== 'inherit' && {
     textAlign: ownerState.align,
```

At one point, I thought, maybe it's to support `<Box sx={{ typography: "inherit", color: "red" }}>subtitle2</Box>` but  this doesn't work

<img width="168" alt="Screenshot 2023-07-19 at 20 15 17" src="https://github.com/mui/material-ui/assets/3165635/f08bffe4-4f44-4b38-b70e-9af8028c57c9">

https://codesandbox.io/s/xenodochial-stitch-wpjcqk?file=/src/Demo.tsx

Reply:
@oliviertassinari You are correct. After giving it some thought, it doesn't make sense to include `variant` in theme's typography. We can also avoid the bypass you mentioned in the sx prop. Looks like a proper solution. 

Also then, like you pointed out
```jsx
<Box sx={{ typography: "inherit", color: "red" }}>subtitle2</Box>
```
should have worked because it is in the theme - https://mui.com/system/typography/#variant. 

However, I'm concerned that this change you proposed might be breaking for developers who are doing this - https://mui.com/material-ui/react-typography/#theme:
```jsx
import * as React from 'react';
import { styled } from '@mui/material/styles';

const Div = styled('div')(({ theme }) => ({
  ...theme.typography.inherit,
  backgroundColor: theme.palette.background.paper,
  padding: theme.spacing(1),
}));

export default function TypographyTheme() {
  return <Div>{"This div's text looks like that of a button."}</Div>;
}
```

Reply:
> I'm concerned that this change you proposed might be breaking for developers who are doing this - https://mui.com/material-ui/react-typography/#theme:

We are not documenting this typography variant, I doubt it would have much impact.

Reply:
To circle back, we fixed this in #38123, all good, much better.